src/pip/_internal/req/req_install.py

```diff
@@ -264,9 +264,21 @@ def populate_link(self, finder, upgrade, require_hashes):
                 package_name=self.name,
                 supported_tags=supported_tags,
             )
+
             if old_link != self.link:
                 logger.debug('Using cached wheel link: %s', self.link)
 
+    @property
+    def from_wheel_cache(self):
+        # type: () -> bool
+        """
+        This function returns whether the file path is in the wheel cache.
+        """
+        if self._wheel_cache is not None and self.link is not None:
+            return bool(self._wheel_cache.cache_dir in self.link.file_path)
+        else:
+            return False
```

Comment:
IMHO this is safer (duck-typing) and clearer:
```suggestion
        try:
            return self._wheel_cache.cache_dir in self.link.file_path
        except AttributeError:
            return False
```

Also please mind the docstring.

Reply:
The AttributeError suggestion is unclear to me. There are too many attribute access and it’s confusing which one causes the exception. The original implementation is more explicit for me (but the `bool()` cast can be removed as suggested).